test/build-with-timeout/build-with-low-timeout.yaml

```diff
@@ -0,0 +1,26 @@
+# Copyright 2018 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: build.knative.dev/v1alpha1
+kind: Build
+metadata:
+  name: test-with-low-timeout
+  labels:
+    expect: failed
```

Comment:
This is probably another case where our integration tests can be improved, but it would be great if we could have `expect: timeout` and have `e2e-test.sh` assert not only that the test failed, but that it failed due to timeout specifically. If the build was changed to `exit 1` the test would still pass. :confused: 

Anyway, not necessary for this PR, but something we should think about for future integration testing improvements.

Reply:
Yeah `failed` is very generic and asserting on specific error would be ideal. I added controller level test for timeout which triggers `syncHandler` twice manually.